File: v1/src/main/java/com/api/v1/medical_slots/services/MedicalSlotRegistrationService.java
package com.api.v1.medical_slots.services;

import com.api.v1.medical_slots.DefaultMedicalSlotResponseDTO;
import org.springframework.http.ResponseEntity;

import java.time.LocalDateTime;

public interface MedicalSlotRegistrationService {
    ResponseEntity<DefaultMedicalSlotResponseDTO> register(String medicalLicenseNumber, LocalDateTime availableAt);
}

File: v2/src/main/kotlin/com/api/v2/medical_slots/services/MedicalSlotRegistrationService.kt
package com.api.v2.medical_slots.services

import com.api.v2.medical_slots.DefaultMedicalSlotResponseDTO
import org.springframework.http.ResponseEntity
import java.time.LocalDateTime

interface MedicalSlotRegistrationService {
    fun register(medicalLicenseNumber: String, availableAt: LocalDateTime): ResponseEntity<DefaultMedicalSlotResponseDTO>
}